### ISSUE TITLE
Higher resolution coastline data for regional maps

### ISSUE THREAD

**Issue by EllieBowler** [icenet-ai/icenet#285]
When producing regional plots for stakeholders, using higher resolution coastline data might be useful so that smaller islands and coastline features can be seen more easily (image below). The cartopy 10m resolution data could be used, or for very small regions higher resolution products such as [open street map land shapefiles](https://osmdata.openstreetmap.de/data/land-polygons.html) could be overlaid using geopandas. 
![image](https://github.com/user-attachments/assets/5448f6e2-d320-49b5-a459-7df586836cd9)

**Comment by EllieBowler** [comment]
Just adding this as we discussed @bnubald- i have code to process and plot regional open street map data for the caribou project, so could look at adding if it turned out to be useful for stakeholders. Otherwise a switch to 10m cartopy coastlines when plotting smaller areas might be useful.

**Comment by bnubald** [comment]
Hi @EllieBowler, if the code is open, could you link it here please?

**Comment by EllieBowler** [comment]
Yes sorry, code to load the osm data for an area of interest and save as a geopandas [here ](https://github.com/icenet-ai/icenet-caribou/blob/main/src/data/load.py#L80). This reads from the [full shapefiles](https://osmdata.openstreetmap.de/data/land-polygons.html) which i saved in a data folder, these are very detailed so takes ages to load the whole thing, which is why i saved regionally. There is an example of plotting one of these geopandas dataframes in combination with xarray [here](https://github.com/icenet-ai/icenet-caribou/blob/main/src/plotting/plotting_utils.py#L60).

**Comment by bnubald** [comment]
Based on some trials, focusing on the Foxe Basin region, comparing a few options, conclusion:

- Using the default `Natural Earth` for pan-Arctic/Antarctic plots.
- `GSHHS` seems like a good one to go for when selecting a particular region and seems to be consistent.

```python
# Foxe basin region
# lon_min, lat_min, lon_max, lat_max
x0, y0, x1, y1 = -96, 61, -66, 70
```

![image](https://github.com/user-attachments/assets/7be0d5f2-22cd-4294-9438-eab65a555fc9)

**Comment by EllieBowler** [comment]
Cool! I can see at this scale for the Foxe Basin the more detailed coastlines are actually a bit harder to see, as there's too much detail around the coast. I wonder how things look in the Coronation Gulf out of interest? It is quite a lot smaller than this, and there are islands which we saw the caribou use for crossing which don't show up in some products. I think the coordinates would be: 
```
# lon_min, lat_min, lon_max, lat_max
x0, y0, x1, y1 = -115, 66, -105, 70
```

**Comment by bnubald** [comment]
Looks comparable to plots generated before (looks like it used the cartopy default):

![image](https://github.com/user-attachments/assets/80a3da86-7203-4226-bf5d-77af5fa159b5)

https://www.bas.ac.uk/project/ai-for-smart-conservation/

**Comment by EllieBowler** [comment]
I should actually update the map on that page - you can see on the animations in the same link it's higher resolution (the OSM data) https://www.bas.ac.uk/project/ai-for-smart-conservation/. But i think the GSHHS/NOAA looks best here too!
